src/compute-types/src/dataflows.rs

```diff
@@ -37,21 +38,33 @@ include!(concat!(env!("OUT_DIR"), "/mz_compute_types.dataflows.rs"));
 
 /// A description of a dataflow to construct and results to surface.
 #[derive(Clone, Debug, Serialize, Deserialize, Eq, PartialEq)]
-pub struct DataflowDescription<P, S: 'static = (), T = mz_repr::Timestamp> {
+pub struct DataflowDescription<
+    P,
+    S: 'static + Serialize + DeserializeOwned = (),
```

Comment:
I actually don't understand why the addition of

```Rust
    #[serde(serialize_with = "mz_ore::serde::map_key_to_string")]
    #[serde(deserialize_with = "mz_ore::serde::string_key_to_btree_map")]
```

requires these annotations. The type already derived `Serialize` and `Deserialize`.